src/RealFourierArray.jl

```diff
@@ -0,0 +1,91 @@
+# Lazy structure for an array whose real and fourier part can be accessed by A.real and A.fourier
+# For normalization conventions, see PlaneWaveBasis.jl
+
+import Base.getproperty, Base.setproperty!, Base.fieldnames
+import Base.*, Base.+, Base./
+import Base.eltype
+
+## these functions are minimally optimized for the moment
+
+## TODO this always allocates both real and fourier parts, which might be a bit wasteful
+## TODO implement broadcasting. This seems to be non-trivial, so I just implement a few simple methods
+## TODO implement compressed representation in Fourier space of real arrays
+
+# Tr is the underlying real type
+# T is the type of the array in real space
+# TODO make this concrete by parametrizing on the type of the arrays
+mutable struct RealFourierArray{Tr <: Real, T <: Union{Tr, Complex{Tr}}}
```

Comment:
Not a fan of `Tr`. `Treal` is not that much longer, but clearer.

Reply:
agreed